mindmeld/components/question_answerer.py

```diff
@@ -2741,11 +2789,122 @@ def is_vector_field(self):
             return self.type in self.VECTOR_TYPES
 
 
+class QuestionAnswererFactory:
+    """
+    Factory class for creating QuestionAnswerers
+
+    usage
+        >>> question_answerer = QuestionAnswererFactory.create_question_answerer(**kwargs)
+        >>> question_answerer.load_kb(...)
+        >>> question_answerer.get(...) # .get(...) or .build_search(...)
+    """
+
+    @classmethod
+    def create_question_answerer(cls, **kwargs):
```

Comment:
I would rather have a clearer set of optional args here like config and app_path. Ideally, config should be a data class with a fixed set of attributes rather than a dict

Reply:
Corrected the code accordingly. The config here is the QUESTION_ANSWERER_CONFIG type dict passed to the QuestionAnswererFactory directly instead of using the QUESTION_ANSWERER_CONFIG from the config.py file. This is to enable QA instantiation without having to create an app.